components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/connector/EmailVerificationOnUpdateConfigImpl.java

```diff
@@ -0,0 +1,122 @@
+import org.wso2.carbon.identity.governance.IdentityGovernanceException;
+import org.wso2.carbon.identity.governance.common.IdentityConnectorConfig;
+import org.wso2.carbon.identity.recovery.IdentityRecoveryConstants;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ * Class which contains config to enable for email verification on update feature.
+ */
+public class EmailVerificationOnUpdateConfigImpl implements IdentityConnectorConfig {
+
+    private static final String CONNECTOR_NAME = "user-claim-update";
+    private static final String CATEGORY = "Account Management Policies";
+    private static final String FRIENDLY_NAME = "User Claim Update";
+    private static final String SUB_CATEGORY = "DEFAULT";
+
+    @Override
+    public String getName() {
+
+        return CONNECTOR_NAME;
+    }
+
+    @Override
+    public String getFriendlyName() {
+
+        return FRIENDLY_NAME;
+    }
+
+    @Override
+    public String getCategory() {
+
+        return CATEGORY;
+    }
+
+    @Override
+    public String getSubCategory() {
+
+        return SUB_CATEGORY;
+    }
+
+    @Override
+    public int getOrder() {
+
+        return 0;
+    }
+
+    @Override
+    public Map<String, String> getPropertyNameMapping() {
+
+        Map<String, String> nameMapping = new HashMap<>();
+        nameMapping.put(IdentityRecoveryConstants.ConnectorConfig.ENABLE_EMAIL_VERIFICATION_ON_UPDATE,
+                "Enable User Email Verification On Update");
+        return nameMapping;
+    }
+
+    @Override
+    public Map<String, String> getPropertyDescriptionMapping() {
+
+        Map<String, String> descriptionMapping = new HashMap<>();
+        descriptionMapping.put(IdentityRecoveryConstants.ConnectorConfig.ENABLE_EMAIL_VERIFICATION_ON_UPDATE,
+                "Trigger a verification notification when user's email address is updated.");
+        return descriptionMapping;
+    }
+
+    @Override
+    public String[] getPropertyNames() {
+
+        List<String> properties = new ArrayList<>();
+        properties.add(IdentityRecoveryConstants.ConnectorConfig.ENABLE_EMAIL_VERIFICATION_ON_UPDATE);
+        return properties.toArray(new String[0]);
+    }
+
+    @Override
+    public Properties getDefaultPropertyValues(String tenantDomain) {
+
+        String enableEmailVerificationOnUpdate = "false";
+        String emailVerificationOnUpdateProperty = IdentityUtil.getProperty(
+                IdentityRecoveryConstants.ConnectorConfig.ENABLE_EMAIL_VERIFICATION_ON_UPDATE);
+
+        if (StringUtils.isNotEmpty(emailVerificationOnUpdateProperty)) {
+            enableEmailVerificationOnUpdate = emailVerificationOnUpdateProperty;
+        }
+
+        Map<String, String> defaultProperties = new HashMap<>();
+        defaultProperties.put(IdentityRecoveryConstants.ConnectorConfig.ENABLE_EMAIL_VERIFICATION_ON_UPDATE,
+                enableEmailVerificationOnUpdate);
```

Comment:
Why we need to instantiate a hashmap as the property is being added for a new property object.
We could add it directly right

Reply:
added it directly.